ml_model/model/results/compare_results.py

```diff
@@ -0,0 +1,172 @@
+            legend_elements = [Line2D([0], [0], marker='o', color='w', label='mean', markerfacecolor='r', markersize=10)]
+            plt.legend(handles=legend_elements, loc='upper right')
+            plt.savefig(ROOT_PATH + 'results/final-boxplots/' + target + '_' + model_name + '_' + metric +'_boxplot_'+ str(SMOTE_TAG) + '.png')
+            plt.close() 
+
+        with open(ROOT_PATH + 'results/mean_median_'+ str(SMOTE_TAG) + '.csv', 'a') as csv_file:
+            writer = csv.DictWriter(csv_file, fieldnames=header.keys())
+            writer.writerow(header)
+
+
+## draw histogram - to check the normality for diff_f1, diff_precision, diff_recall
+columns = ['diff_precision','diff_recall','diff_f1', 'diff_auc']
+for column in columns:
+    hist = overall_df.hist(column=column)
+    plt.savefig(ROOT_PATH + 'results/histograms/final_hist_' + column +'_'+ str(SMOTE_TAG) + '.png')
+
+## draw histogram for all the input features in a single plot
+df_raw_features = pd.read_csv(ROOT_PATH + 'data/understandability_with_warnings.csv')
+with open(ROOT_PATH + "classification/experiments.jsonl") as jsonl_file:
+    experiments = [json.loads(jline) for jline in jsonl_file.read().splitlines()]
+    for experiment in experiments:
+        full_dataset = df_raw_features.dropna(subset=experiment["target"])
+        feature_X_c = full_dataset[experiment["features"]].iloc[:, 1:]
+        ## draw histogram with labels
+        names = list(feature_X_c.columns)
+        feature_X_c.hist(figsize=(10, 6), bins=len(feature_X_c.columns),  stacked=True)
+        plt.suptitle('')
+        plt.tight_layout()
+        plt.savefig(ROOT_PATH + 'results/histograms/final_hist_code_features'+ experiment["target"] +'_'+ str(SMOTE_TAG) + '.png')
+        plt.close()
+
+#######################
+## STATISTICAL TESTS ##
+#######################
+tests = {'metric':'', 'model':'', 'target':'', 'wilcoxon_test(p-value)':''}
+## write header
+with open(ROOT_PATH + 'results/final_statistics_'+ str(SMOTE_TAG) + '.csv', "w") as csv_file:
+    writer = csv.DictWriter(csv_file, fieldnames=tests.keys())
+    writer.writeheader()
+
+metrics = {'diff_f1': ['f1_c','f1_cw'], 
+           'diff_precision': ['precision_c','precision_cw'], 
+           'diff_recall': ['recall_c','recall_cw'],
+           'diff_auc': ['auc_c','auc_cw']}
+
+model_names = ["SVC", "knn_classifier", "logistic_regression", "randomForest_classifier", "mlp_classifier"]
+targets = ['ABU50', 'BD50', 'PBU']
+
+for model_name in models:
+    for target in targets:
+        model_data = overall_df.query("model == '" + model_name + "' and target == '" + target + "'")
+        print (model_data.shape[0])
+        for metric in metrics.keys():
+            x = model_data[metrics[metric][0]] ## code
+            y = model_data[metrics[metric][1]] ## code+warnings
+
+            ## 1. WILCOXON TEST ##
+            ## null hypothesis: There is no statistically significant difference between x and y
```

Comment:
small fixes: 
1. x=code and y=code+warnings
2. H0: x>y (fix the description: the performance (based on a metric) of the model using code features is greater than the performance when using code+warnings)